### PR TITLE
add some tests for if empty space is present

### LINKED ISSUE #62140: aurora caelus can still sometimes happen on icebox
## Issue:

aurora caelus can still sometimes happen on IceBox. IceBox doesn't have any `/area/space/nearstation` or `/area/solars`, the only areas which have the `AREA_USES_STARLIGHT` flag that aurora caelus uses.

## Reproduction:

Wait around long enough on icebox with random events enabled

### PR DESCRIPTION
**Pull request by Mooshimi**
## About The Pull Request

Fixes: #62140
Fixes part of: #53689
## Why It's Good For The Game

Aurora Caelus should no longer run on icebox as intended.
Station goal on stations without space shouldn't get station goals that require space.

## Changelog

:cl:
fix: Added a test in Aurora Caelus event for if the station has the ability to show it.
fix: Stations without space should no longer receive station objectives that require space to be present.
/:cl:

### CODE REVIEW

**Comment by Crushtoe** [comment]
counter: aurora will still show up in the kitchen even on icebox

**Review by MrMelbert — CHANGES_REQUESTED** [review]

File: code/modules/station_goals/shield.dm [hunk 1]
```diff
@@ -4,6 +4,8 @@
 /datum/station_goal/station_shield
 	name = "Station Shield"
 	var/coverage_goal = 500
+	/// requires space, 1 means yes
+	requires_space = 1
```
Comment (ShizCalev) [inline code/modules/station_goals/shield.dm:8]: it's 2022

File: code/modules/station_goals/station_goal.dm [hunk 2]
```diff
@@ -5,6 +5,7 @@ GLOBAL_LIST_EMPTY_TYPED(station_goals, /datum/station_goal)
 	var/name = "Generic Goal"
 	var/weight = 1 //In case of multiple goals later.
 	var/required_crew = 10
+	var/requires_space = 0 //if the goal requires the station to be in space. 0 = no
```
Comment (ShizCalev) [inline code/modules/station_goals/station_goal.dm:8]: i am judging you.

File: code/game/gamemodes/game_mode.dm [hunk 3]
```diff
@@ -185,6 +185,11 @@
 /datum/game_mode/proc/generate_station_goals(greenshift)
 	var/goal_budget = greenshift ? INFINITY : CONFIG_GET(number/station_goal_budget)
 	var/list/possible = subtypesof(/datum/station_goal)
+	if(!(SSmapping.empty_space))
+		for(var/datum/station_goal/goal in possible)
+			if(goal.requires_space == 1)
```
Comment (ShizCalev) [inline code/game/gamemodes/game_mode.dm:190]: is there a requires == 2?

**Review by ShizCalev — CHANGES_REQUESTED** [review]

File: code/game/gamemodes/game_mode.dm (hunk 3 above)
Comment (Mooshimi) [reply code/game/gamemodes/game_mode.dm:190]: no, it's just 1 if it does require space, but if it bothers you I can just change it to TRUE/FALSE

File: code/modules/station_goals/station_goal.dm (hunk 2 above)
Comment (Mooshimi) [reply code/modules/station_goals/station_goal.dm:8]: this comment is vague and unhelpful

File: code/modules/station_goals/shield.dm (hunk 1 above)
Comment (Mooshimi) [reply code/modules/station_goals/shield.dm:8]: vague and unhelpful

Comment (Mothblocks) [reply code/modules/station_goals/shield.dm:8]: theyre vaguely telling you to use TRUE
